src/chttpd_sup.erl

```diff
@@ -57,10 +54,8 @@ lru_opts() ->
             []
     end.
 
-auth_cache_handler() ->
-    case application:get_env(chttpd, auth_cache) of
-        {ok, Module} ->
-            Module;
-        _ ->
-            chttpd_auth_cache
-    end.
+add_auth_cache_if_missing(Children) ->
```

Comment:
this is in case `couch_epi:register_service(chttpd_epi)` returns a replacement but it's ad-hoc. Would it a better api to pass the pre-epi children in and then use the return value instead? that way chttpd_epi plugins could remove unwanted children cleanly. something like;

```
Children0 = [?CHILD(auth_cache_handler(), worker), and others...],
Children1 = couch_epi:register_service(chttpd_epi, Children0),
{ok, {{one_for_one, 3, 10}, Children1}.
```
register_service would modify the input list, replacing the auth_cache childspec.